ts/store/helpers/indexer.ts

```diff
@@ -1,44 +1,48 @@
 import { entries } from "lodash";
 
-/**
- * IndexedObject helps creating objects that
- * are indexed by their id (storing both
- * byId and allIds)
- */
-
-type Id = string | number;
-
-// an object with an identifier
-// (assuming strings and numbers
-// only as identifiers -- change as needed)
-interface WithId {
-  id: Id;
+interface IStringKeyedObject {
```

Comment:
you may want to use a StrMap from fp-ts which has some nice mapping methods (ie. collect())

ref. https://github.com/gcanti/fp-ts/blob/master/src/StrMap.ts

Reply:
Using `StrMap` instead of `IStringKeyedObject` would be a bit of an overkill, since the only operation that is done on those object is extracting the id (`obj.id`). 

Additionally, wouldn't `keyof` not work as expected with `StrMap<A>`, since `value` is typed to be [`{ [key: string]: A }`](https://github.com/gcanti/fp-ts/blob/75598e23db6af1049e1a9e9fbfe4ab8737ad8da1/src/StrMap.ts#L38)? This would allow passing any string key -- even those not existing for the specified type.

Reply:
btw I'm planning to remove lodash from all our projects, so any alternative is good :D